### PR TITLE
Allow an empty array of BypassActors in Ruleset struct in CreateRuleset endpoint

### PR DESCRIPTION
**Pull request by Matthew-Reidy**
Fixes: [#3137](https://github.com/google/go-github/issues/3137)

added an extra Ruleset struct and CreateRuleset endpoint in repo_rules.go to allow the BypassActors field to be passed as nil or an empty array

### CODE REVIEW

**Review by gmlewis — CHANGES_REQUESTED** [review]
It think it might be a more pleasant user experience to not introduce a new struct,
but just a new endpoint that still uses the `Ruleset` struct as input, but internally uses a non-exported struct like the one you created. 

To do this, you would need to copy over the fields to this non-exported struct and then send that instead of the original Ruleset.

Also, let's please not start using `BPActr` as a short abbreviation for `BypassActor`, as we try to make this repo as searchable as possible, and nobody is going to guess `bpactr` as a search string.

Also, all new godoc-style comments should be complete sentences with ending periods. Thanks.

**Comment by Matthew-Reidy** [comment]
Do you mean something like having an anonymous struct identical to Ruleset in the new functions scope that would just copy over the values of Ruleset to?

**Comment by gmlewis** [comment]
> Do you mean something like having an anonymous struct identical to Ruleset in the new functions scope that would just copy over the values of Ruleset to? I understand your idea but i'm confused about the implementation.

Yes, that is what I mean.

We have other examples of this in our repo.

Here is one of many: https://github.com/google/go-github/blob/master/github/repos.go#L546-L573

I found this by typing:

```bash
$ grep 'type [a-z]' github/*.go
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:556]:
```suggestion
	rsNoBypassActor := &RulesetNoOmitBypassActors{}

	if rs != nil {
		rsNoBypassActor = &RulesetNoOmitBypassActors{
			ID:           rs.ID,
			Name:         rs.Name,
			Target:       rs.Target,
			SourceType:   rs.SourceType,
			Source:       rs.Source,
			Enforcement:  rs.Enforcement,
			BypassActors: rs.BypassActors,
			NodeID:       rs.NodeID,
			Links:        rs.Links,
			Conditions:   rs.Conditions,
			Rules:        rs.Rules,
		}
	}
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:399]:
```suggestion
// rulesetNoOmitBypassActors represents a GitHub ruleset object. The struct does not omit bypassActors if the field is nil or an empty array is passed.
type rulesetNoOmitBypassActors struct {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Yes, this is what I was talking about - the only difference is that the new struct should not be exported because the end-user never has know or worry about it.

**Comment by Matthew-Reidy** [comment]
Ok. I commited your suggested changes. I didn't know rulesetNoOmitBypassActors made it non exported. Learned something new about the language. Thanks!

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:538]:
```suggestion
	rsNoBypassActor := &rulesetNoOmitBypassActors{}
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:541]:
```suggestion
		rsNoBypassActor = &rulesetNoOmitBypassActors{
```

**Comment by gmlewis** [comment]
Please re-run step 4 in CONTRIBUTING.md to re-generate the auto-generated files... then the tests should all pass.
You can also test locally before you push by running `go test ./...`

**Review by gmlewis — APPROVED** [review]
Thank you, @Matthew-Reidy !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos_rules.go
Comment (tomfeigin) [inline github/repos_rules.go:399]: I think that you can embed the entire `Ruleset` object and shadow just the `bypass_actors`

```golang
type rulesetNoOmitBypassActors struct {
   Ruleset
   BypassActors []*BypassActor `json:"bypass_actors"`
}
```

example: https://goplay.tools/snippet/7lHZci7LW0y

@gmlewis WDYT?

Then you can use it like so:

```golang
func (s *RepositoriesService) UpdateRulesetNoBypassActor(...rs *Ruleset...
  rsNoBypassActor := &rulesetNoOmitBypassActors{}
   if rs != nil {
      rsNoBypassActor := &rulesetNoOmitBypassActors{Ruleset: *rs}
   }
```

File: github/repos_rules.go
Comment (tomfeigin) [inline github/repos_rules.go:530]: NIT: `marschall` -> `marshal`

File: github/repos_rules.go
Comment (gmlewis) [reply github/repos_rules.go:399]: Sounds great, @tomfeigin - as long as we have a unit test that demonstrates this behavior (meaning a `BypassActors=nil` still gets passed through in JSON as `"bypass_actors":[]`.

**Comment by lewismiddleton** [comment]
This would be useful for me as I'm tracking [a bug](https://github.com/integrations/terraform-provider-github/issues/2269) in the GitHub terraform provider that I think links to #3137.

@Matthew-Reidy are you planning on picking this up again?

**Comment by Matthew-Reidy** [comment]
I can fix the spelling mistake but as for the other proposed change whats the justification? The solution in place now on this PR seems to do the same things and has testing evidence behind it. Thanks.

**Review by tomfeigin — APPROVED** [review]
My comments are only relevant for styling, let's merge it

**Comment by gmlewis** [comment]
Thank you, @tomfeigin and @Matthew-Reidy !
@tomfeigin - after this is merged, feel free to create a PR to improve the styling if you wish.

Merging.

**Pull request merged**
